packages/@lwc/template-compiler/src/shared/types.ts

```diff
@@ -105,7 +105,7 @@ export interface IRComment extends IRBaseNode<parse5.CommentNode> {
 
 export type IRNode = IRComment | IRElement | IRText;
 
-export const enum IRAttributeType {
+export enum IRAttributeType {
```

Comment:
@ekashida `const enum` is inlined and removed by the time the code hits runtime. This is fine if the enum is only used inside the same file, since the compiler will inline the enum, but not if the type is referenced elsewhere.

In this case, `IRAttributeType` is exported by the template-compiler and is used by lwc-platform code. The lwc-platform code now sees `undefined` at runtime for `IRAttributeType` instead of the expected enum.

Reply:
Note that this change is needed to fix a regression in `lwc-platform`.

Reply:
`const enum` also isn't providing much benefit here. The main benefit is a smaller bundle size, and we don't care for the `template-compiler` because it runs in Node.